scopes/html/utils.ts

```diff
@@ -0,0 +1,46 @@
+import { HtmlComposition } from './interfaces';
+
+export async function fetchHtmlFromUrl(url: string) {
+    return fetch(url)
+        .then(response => response.text())
+        .then(data => data)
+}
+
+export function createElementFromString(htmlString: string) {
+    const htmlFragment = document.createRange().createContextualFragment(htmlString);
+    return htmlFragment;
+}
+
+export function renderTemplate(root: HTMLElement, template: string) {
+    root.appendChild(createElementFromString(template));
+}
+
+/**
+ * this mounts compositions into the DOM in the component preview.
+ * this function can be overridden through ReactAspect.overrideCompositionsMounter() API
+ * to apply custom logic for component DOM mounting.
+ */
+export const RenderHtmlComposition = (target: HTMLElement | null, composition: HtmlComposition) => {
+    if (!target) return;
+
+    // first clear the root node from any previous compositions. Required as all compositions
+    // of a specific component are rendered in the same iframe
+    target.innerHTML = '';
```

Comment:
might not want to reset HTML when composition is a function? i.e. if they want to update the DOM themselves

Reply:
would have to do it when changing between compositions, though